src/Proto.Actor/EventStream/DeadLetter.cs

```diff
@@ -45,7 +45,9 @@ public DeadLetterProcess(ActorSystem system) : base(system)
 
         protected internal override void SendUserMessage(PID pid, object message)
         {
-            System.Metrics.Get<ActorMetrics>().DeadletterCount.Inc(new[] {System.Id, System.Address, message.GetType().Name});
+            if (!System.Metrics.IsNoop)
+                ActorMetrics.DeadletterCount.Add(1, new("id", System.Id), new("address", System.Address), new("messagetype", message.GetType().Name));
```

Comment:
This could end up with messagetype 'MessageEnvelope', which is probably not what we want :)

Reply:
Added message unwrapping in user message handler. AFAIK system messages will not be wrapped?

Reply:
System messages will not be wrapped, but that is handled by MessageEnvelope.Unwrap. If the message is not a MessageEnvelope, that will just return the exact same message as was passed in